### PR TITLE
Add Django QS 01-login

### PR DESCRIPTION
**Pull request by alexisluque**

### CODE REVIEW

**Review by aaguiarz — APPROVED** [review]

File: articles/quickstart/webapp/django/01-login.md [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+  ]
+}) %>
+
+<%= include('../_includes/_getting_started', { library: 'Django', callback: 'http://localhost:8000/complete/auth0' }) %>
+
+This guide will use [`social_django`](https://github.com/python-social-auth/social-app-django) which is the Django implementation of [Python Social Auth](http://python-social-auth.readthedocs.io/en/latest/). It adds an OAuth stack to the [user authentication & authorization system](https://docs.djangoproject.com/en/1.11/topics/auth/) bundled by the Django Web Framework.
+
+## Install the Dependencies
+
+Add the following dependencies to your `requirements.txt` and run `pip install -r requirements.txt`
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: I might break this out:

> Add the following dependencies to your `requirements.txt`:


and then after the code block, follow it with:

> Once the dependencies are listed in `requirements.txt`, run the following command:

```
pip install -r requirements.txt
```

That way the section flows sequentially - add this stuff, then run this command. Rather than telling them "you'll need to run a command, but after you fill out the requirements".

File: articles/quickstart/webapp/django/01-login.md [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+
+```text
+django
+social-auth-app-django
+python-jose
+six
+```
+
+## Create a Django project
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Most of your H2s are capitalized, so keep the pattern (Create a Django Project)

File: articles/quickstart/webapp/django/01-login.md [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+```python
+# webappexample\settings.py
+
+SOCIAL_AUTH_TRAILING_SLASH = False                    # Remove end slash from routes
+SOCIAL_AUTH_AUTH0_DOMAIN = '${account.namespace}'
+SOCIAL_AUTH_AUTH0_KEY = '${account.clientId}'
+SOCIAL_AUTH_AUTH0_SECRET = '${account.clientSecret}'
+```
+
+Set the `SOCIAL_AUTH_AUTH0_SCOPE` variable with the scopes the application will request when authenticating. Check the [Scopes documentation](scopes/current) for more information.
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Missing starting slash: `[Scopes documentation](/scopes/current)`

File: articles/quickstart/webapp/django/01-login.md [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+
+SOCIAL_AUTH_AUTH0_SCOPE = [
+    'openid',
+    'profile'
+]
+```
+
+## Initialize the Database
+
+The `social_django` application defined in `INSTALLED_APPS` requires a database. Run the following command to create all the required databases for the applications defined in  `INSTALLED_APPS`:
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Extra space before `INSTALLED_APPS` here:

> ...defined in `INSTALLED_APPS`

File: articles/quickstart/webapp/django/01-login.md [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+```bash
+$ python manage.py migrate
+```
+
+## Create the Auth0 Authentication Backend
+
+Up to now we configured `social_django`. The next step is to create an Authentication Backend that bridges `social_django` with Auth0.
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Maybe reword: 

> The `social_django` application is now configured. The next step is to create an authentication backend that bridges `social_django` with Auth0.

File: articles/quickstart/webapp/django/01-login.md [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+                'first_name': payload['name'],
+                'picture': payload['picture'],
+                'user_id': payload['sub']}
+```
+
+::: note
+The callback URL will be calculated by `social-auth` by concatenating `/callback` with the backend `name` property, so it will be `/callback/auth0`.
+:::
+
+Register the authentication backends in `settings.py`. Add the custom backend for `Auth0` and `ModelBackend` to users be able to login with username/password method.
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Is this 

> to users be able to login

meant to be 

> for users to be able to login

or something else?

File: articles/quickstart/webapp/django/01-login.md [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+        'picture': auth0user.extra_data['picture']
+    }
+    
+    return render(request, 'dashboard.html', {
+        'auth0User': auth0user,
+        'userdata': json.dumps(userdata, indent=4)
+    })
+```
+
+Add the following snippet to `dashboard.html` to display the [user information](https://auth0.com/docs/user-profile/normalized/oidc).
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: Use internal URL: `[user information](/user-profile/normalized/oidc)`

File: articles/quickstart/webapp/django/01-login.md [hunk 8]
```diff
@@ -0,0 +1,268 @@
...
+    <h1 id="logo"><img src="//cdn.auth0.com/samples/auth0_logo_final_blue_RGB.png" /></h1>
+    <img class="avatar" src="{{ auth0User.extra_data.picture }}"/>
+    <h2>Welcome {{ user.username }}</h2>
+    <pre>{{ userdata }}</pre>
+</div>
+```
+
+## Logout
+
+To logout a user, add a link to `/logout` URL in `dashboard.html`.
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: This sounds better written 

> To log a user out, add a link to `/logout` in `dashboard.html`.

File: articles/quickstart/webapp/django/01-login.md [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+</div>
+```
+
+## Add URL Mappings
+
+In previous steps we added methods to the `views.py` file. We need to map those methods to URLs.
+
+Django has a [URL dispatcher](https://docs.djangoproject.com/en/1.11/topics/http/urls/) that lets you map URL patterns to views.
+
+Add mappins for the root folder, the dashboard folder and the authentication applications in `urls.py`.
```
Comment (jeffreylees) [inline articles/quickstart/webapp/django/01-login.md]: "Mappings" typo and Oxford comma

> Add mappings for the root folder, the dashboard folder, and the authentication applications in `urls.py`.

**Review by jeffreylees — CHANGES_REQUESTED** [review]
Just some quick copyedits

**Review by jeffreylees — APPROVED** [review]
